### ISSUE TITLE
Ошибка при запуске docker-compose up -d

### ISSUE THREAD

**Issue by mpvfaq** [nixys/nxs-support-bot#14]
docker-compose up -d
ERROR: The Compose file './docker-compose.yml' is invalid because:
Unsupported config option for configs: 'app.conf'
Unsupported config option for networks: 'docker-lan'
Unsupported config option for volumes: 'mysql'
Unsupported config option for x-default: 'networks'
Unsupported config option for services: 'supportbot'

**Comment by borisershov** [comment]
Приветствую! Покажите, пожалуйста, содержимое вашего файла `docker-compose.yml`

**Comment by mpvfaq** [comment]
x-default: &default
  env_file:
  - .env
  networks:
  - docker-lan
services:
  redis:
    container_name: redis
    restart: always
    image: bitnami/redis:latest
    <<: *default
    volumes:
    - redis:/bitnami/redis/data
  mysql:
    image: percona:8
    restart: always
    container_name: mysql
    <<: *default
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
    volumes:
    - mysql:/var/lib/mysql
  migration:
    image: nixyslab/nxs-support-bot-migrations:latest
    container_name: migration
    <<: *default
    command:
    - -database
    - mysql://${MYSQL_USER}:${MYSQL_PASSWORD}@tcp(${MYSQL_HOST}:${MYSQL_PORT})/${MYSQL_DATABASE}?multiStatements=true
    - -path
    - /migrations
    - up
    depends_on:
      mysql:
        condition: service_healthy
  supportbot:
    image: nixyslab/nxs-support-bot:latest
    restart: always
    container_name: supportbot
    <<: *default
    configs:
    - app.conf
    depends_on:
      migration:
        condition: service_completed_successfully

configs:
  app.conf:
    file: ./app.conf

volumes:
  redis:
  mysql:

networks:
  docker-lan: {}

**Comment by borisershov** [comment]
Скорее всего дело в том, что у вас нарушены отступы в `docker-compose.yml`. Дело в том, что формат YAML требователен к пробелам в начале строк и с помощью них отделяет один блок от другого. Вероятно, когда вы копировали содержимое файла из репозитория, то ваш редактор не учёл пробелы, из-за чего вы и получаете ошибку.

Попробуйте либо проконтролировать отступы при копировании содержимого этого файла, либо склонировать репозиторий, а затем отдельно этот файл (в этом случае его содержимое должно скопироваться верно). А уже затем отредактировать параметры под вашу инфраструктуру.

**Comment by mpvfaq** [comment]
Дело в том что я брал исходники отсюда https://github.com/nixys/nxs-support-bot/releases/tag/v1.4.0
Т.к. git clone git@github.com:nixys/nxs-support-bot.git
Выдает следующую ошибку:

Cloning into 'nxs-support-bot'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by borisershov** [comment]
> Дело в том что я брал исходники отсюда https://github.com/nixys/nxs-support-bot/releases/tag/v1.4.0

Да, такой способ получаения `docker-compose.yml` тоже подойдёт в дополнение к двум описанным мной в предыдущем комментарии. Только при копировании, пожалуйста, убедитесь, что в этом файле не нарушены отступы вначале строк.
